src/inc/msquic.h

```diff
@@ -1122,6 +1122,10 @@ typedef struct QUIC_CONNECTION_EVENT {
             uint16_t BidirectionalCount;
             uint16_t UnidirectionalCount;
         } STREAMS_AVAILABLE;
+        struct {
+            BOOLEAN Bidirectional;
+            uint64_t StreamLimit;
```

Comment:
I was thinking about this more... does it make sense to expose the absolute offset (`StreamLimit`) to the app, or should we somehow convert this to a relative value, since that's what our FC works with already?

Reply:
Somehow do something like limit - streams_closed - streams_open

Reply:
If you mean like follow what `STREAMS_AVAILABLE` does, I would say sometimes the `STREAMS_AVAILABLE` confused me as following scenario:

If client starts 2 streams locally while server allows 0, client get event STREAMS_AVAILABLE: 0 but actually (-2)

After server unblock it, lets say it set to max 2, we will still get STREAMS_AVAILABLE: 0.

I personally prefer to change STREAMS_AVAILABLE to absolute value instead.

WDYT?

Reply:
> If client starts 2 streams locally while server allows 0, client get event STREAMS_AVAILABLE: 0 but actually (-2)

IMO, this is a slightly different issue. Perhaps this event could indicate total available, and how many are blocked, or something like that. Either way, I'd suggest that to be (possibly) changed in a separate PR.

> I personally prefer to change STREAMS_AVAILABLE to absolute value instead.

Imagine you're doing 1 app "frame" == 1 QUIC stream (i.e. LOTS of streams per second). After doing this for a few minutes, you're up in the millions of streams that have been opened. Then all of a sudden you happen to get a STREAMS_BLOCKED event with `StreamLimit` set to 1,226,995. How does this number help you, if your current stream limit is 100?

Reply:
magine you're doing 1 app "frame" == 1 QUIC stream (i.e. LOTS of streams per second). After doing this for a few minutes, you're up in the millions of streams that have been opened. Then all of a sudden you happen to get a STREAMS_BLOCKED event with StreamLimit set to 1,226,995. How does this number help you, if your current stream limit is 100?

I don't quit understand the scenario how could one endpoint open 1,226,995 streams while peer only allows 100?

Reply:
If I read RFC9000 correctly, the STREAMS_BLOCKED should have 100, not 1,226,995, right?

Reply:
> I don't quit understand the scenario how could one endpoint open 1,226,995 streams while peer only allows 100?

Not all streams are open in parallel. They are opened and closed immediately, keeping ~100 open at a time. Eventually, after opening and closing enough, perhaps the rate of opening exceeds the rate at which the receiver can handle; a queue builds up and the opener becomes "blocked".